### PR TITLE
fix recursive macro expansion check

### PR DESCRIPTION
**Pull request by WalterBright**
In porting this to D, I noticed the recursive expansion check wasn't working, but the code worked anyway because there's an overall nesting limit. The result was this could cause things to run slowly.

Also removed long dead code.

### CODE REVIEW

File: src/macro.c [hunk 1]
```diff
@@ -379,25 +364,29 @@ void Macro::expand(OutBuffer *buf, size_t start, size_t *pend,
...
-#endif
                     if (m->inuse && marglen == 0)
                     {   // Remove macro invocation
                         buf->remove(u, v + 1 - u);
                         end -= v + 1 - u;
                     }
-                    else if (m->inuse && arglen == marglen && memcmp(arg, marg, arglen) == 0)
+                    else if (m->inuse &&
+                             ((arglen == marglen && memcmp(arg, marg, arglen) == 0) ||
+                              // marg is arg with blue paint added
```
Comment (yebblies) [inline src/macro.c]: Please no comments inside expressions.

File: src/macro.c [hunk 2]
```diff
@@ -379,25 +364,29 @@ void Macro::expand(OutBuffer *buf, size_t start, size_t *pend,
...
+                               marg[1] == '{' &&
+                               memcmp(arg, marg + 2, arglen) == 0 &&
+                               marg[marglen - 2] == 0xFF &&
+                               marg[marglen - 1] == '}'
+                              )
+                             )
+                            )
                     {   // Recursive expansion; just leave in place
-
+                        ;
```
Comment (yebblies) [inline src/macro.c]: Why?  This will actually generate a warning in D.

Comment (WalterBright) [inline src/macro.c]: No, it won't. What it does do is make the line 'executable' so the -cov analyzer will indicate whether it was executed or not.

Comment (yebblies) [inline src/macro.c]: Yes, it will.  `Warning: use '{ }' for an empty statement, not a ';'`

File: src/macro.c [hunk 3]
```diff
@@ -120,15 +111,9 @@ size_t extractArgN(const utf8_t *p, size_t end, const utf8_t **pmarg, size_t *pm
...
   Largstart:
-#if 1
     // Skip first space, if any, to find the start of the macro argument
     if (n != 1 && v < end && isspace(p[v]))
         v++;
-#else
-    // Skip past spaces to find the start of the macro argument
-    for (; v < end && isspace(p[v]); v++)
-        ;
-#endif
```
Comment (andralex) [inline src/macro.c]: all hail the red

**Comment by andralex** [comment]
unittests?

**Comment by andralex** [comment]
oh it's speed

**Pull request merged**